gunicorn/reloader.py

```diff
@@ -68,48 +70,46 @@ def run(self):
 
 if has_inotify:
 
-    class InotifyReloader(threading.Thread):
+    class InotifyReloader(ReloaderBase):
         event_mask = (inotify.constants.IN_CREATE | inotify.constants.IN_DELETE
                       | inotify.constants.IN_DELETE_SELF | inotify.constants.IN_MODIFY
                       | inotify.constants.IN_MOVE_SELF | inotify.constants.IN_MOVED_FROM
                       | inotify.constants.IN_MOVED_TO)
 
         def __init__(self, extra_files=None, callback=None):
-            super().__init__()
-            self.setDaemon(True)
-            self._callback = callback
+            super().__init__(extra_files=extra_files, callback=callback)
             self._dirs = set()
             self._watcher = Inotify()
 
-            for extra_file in extra_files:
-                self.add_extra_file(extra_file)
-
         def add_extra_file(self, filename):
             dirname = os.path.dirname(filename)
 
-            if dirname in self._dirs:
-                return
+            with self._extra_files_lock:
```

Comment:
This lock was removed in #2134 because the reloader starts after the application has been loaded now.